Hmm tests fail for me on nixos x86_64:

```
FAILED test/test_poolmanager.py::TestPoolManager::test_deprecated_no_scheme - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='evil.com', port=...
FAILED test/with_dummyserver/test_socketlevel.py::TestSocketClosing::test_connection_cleanup_on_read_timeout - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port...
FAILED test/with_dummyserver/test_socketlevel.py::TestSSL::test_ssl_custom_validation_failure_terminates - pytest.PytestUnhandledThreadExceptionWarning: Exception in thread Thread-137
FAILED test/with_dummyserver/test_socketlevel.py::TestHeaders::test_headers_sent_with_add[GET-None] - AssertionError: assert b'A: 1\r\nA: 4\r\nC: 3, 5\r\nC: 6\r\nB: 2\r\nB: 3' i...
FAILED test/with_dummyserver/test_socketlevel.py::TestStream::test_large_compressed_stream - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[file-POST-False] - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[file-PUT-True] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[file-PATCH-True] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[file-PATCH-False] - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[generator-POST-True] - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[generator-PUT-False] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[generator-PATCH-True] - AssertionError: assert b'\r\n\r\na\r\nxxxxxxxxxx\r\n0\r\n\r\n' in b'PATCH /...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[generator-PATCH-False] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[bytes-PUT-True] - urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetEr...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_specified[bytes-PATCH-True] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[file-PUT] - urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetEr...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[file-PATCH] - urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetEr...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[generator-POST] - AssertionError: assert b'\r\n\r\na\r\nxxxxxxxxxx\r\n0\r\n\r\n' in b'POST / ...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[generator-PUT] - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[generator-PATCH] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[bytes-POST] - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[bytes-PUT] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[bytearray-POST] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[file_text-POST] - AssertionError: assert 0 == 5
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_chunked_not_specified[file_text-PUT] - AssertionError: assert b'\r\n\r\na\r\nxxxxxxxxxx\r\n0\r\n\r\n' in b'PUT / H...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_framing_set_via_headers[content-length-10-: 10\r\n\r\nxxxxxxxx-lower] - urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', p...
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_framing_set_via_headers[transfer-encoding-chunked-: chunked\r\n\r\n8\r\nxxxxxxxx\r\n0\r\n\r\n-lower] - AssertionError: assert False
FAILED test/with_dummyserver/test_socketlevel.py::TestContentFraming::test_framing_set_via_headers[transfer-encoding-chunked-: chunked\r\n\r\n8\r\nxxxxxxxx\r\n0\r\n\r\n-title] - urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetEr...

```

--- a/pkgs/development/python-modules/urllib3/default.nix
+++ b/pkgs/development/python-modules/urllib3/default.nix
@@ -1,4 +1,5 @@
 { lib
+, backports-zoneinfo
 , brotli
 , brotlicffi
 , buildPythonPackage
@@ -8,18 +9,16 @@
 , hatchling
 , idna
 , isPyPy
-, mock
 , pyopenssl
 , pysocks
-, pytest-freezegun
 , pytest-timeout
 , pytestCheckHook
-, python-dateutil
+, pythonOlder
 , tornado
 , trustme
 }:
 
-buildPythonPackage rec {
+let self = buildPythonPackage rec {
   pname = "urllib3";
   version = "2.0.7";
   pyproject = true;
@@ -31,41 +30,6 @@ buildPythonPackage rec {
 
   nativeBuildInputs = [
     hatchling
-  ];
-
-  # FIXME: remove backwards compatbility hack
-  propagatedBuildInputs = passthru.optional-dependencies.brotli
-    ++ passthru.optional-dependencies.socks;
-
-  nativeCheckInputs = [
-    python-dateutil
-    mock
-    pytest-freezegun
-    pytest-timeout
-    pytestCheckHook
-    tornado
-    trustme
-  ];
-
-  # Tests in urllib3 are mostly timeout-based instead of event-based and
-  # are therefore inherently flaky. On your own machine, the tests will
-  # typically build fine, but on a loaded cluster such as Hydra random
-  # timeouts will occur.
-  #
-  # The urllib3 test suite has two different timeouts in their test suite
-  # (see `test/__init__.py`):
-  # - SHORT_TIMEOUT
-  # - LONG_TIMEOUT
-  # When CI is in the env, LONG_TIMEOUT will be significantly increased.
-  # Still, failures can occur and for that reason tests are disabled.
-  doCheck = false;
-
-  preCheck = ''
-    export CI # Increases LONG_TIMEOUT
-  '';
-
-  pythonImportsCheck = [
-    "urllib3"
   ];
 
   passthru.optional-dependencies = {
@@ -86,11 +50,44 @@ buildPythonPackage rec {
     ];
   };
 
+  nativeCheckInputs = [
+    pytest-timeout
+    pytestCheckHook
+    tornado
+    trustme
+  ] ++ lib.optionals (pythonOlder "3.9") [
+    backports-zoneinfo
+  ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
+
+  # Tests in urllib3 are mostly timeout-based instead of event-based and
+  # are therefore inherently flaky. On your own machine, the tests will
+  # typically build fine, but on a loaded cluster such as Hydra random
+  # timeouts will occur.
+  #
+  # The urllib3 test suite has two different timeouts in their test suite
+  # (see `test/__init__.py`):
+  # - SHORT_TIMEOUT
+  # - LONG_TIMEOUT
+  # When CI is in the env, LONG_TIMEOUT will be significantly increased.
+  # Still, failures can occur and for that reason tests are disabled.
+  doCheck = false;
+
+  passthru.tests.pytest = self.overridePythonAttrs (_: { doCheck = true; });
+
+  preCheck = ''
+    export CI # Increases LONG_TIMEOUT
+  '';
+
+  pythonImportsCheck = [
+    "urllib3"
+  ];
+
   meta = with lib; {
-    description = "Powerful, sanity-friendly HTTP client for Python";
-    homepage = "https://github.com/shazow/urllib3";
+    description = "Powerful, user-friendly HTTP client for Python";
+    homepage = "https://github.com/urllib3/urllib3";
     changelog = "https://github.com/urllib3/urllib3/blob/${version}/CHANGES.rst";
     license = licenses.mit;
     maintainers = with maintainers; [ fab ];
   };
-}
+};
+in self

--- a/pkgs/development/python-modules/urllib3/default.nix
+++ b/pkgs/development/python-modules/urllib3/default.nix
@@ -21,12 +21,12 @@
 
 buildPythonPackage rec {
   pname = "urllib3";
-  version = "2.0.6";
-  format = "pyproject";
+  version = "2.0.7";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-sZ4ahdIGtW198dXmg99KdyUlKpZOOZNkjdD7WhwVdWQ=";
+    hash = "sha256-yX394fe9Q6ccjSpY42npsr9pLRM06p+crlWt19DdD4Q=";
   };
 
   nativeBuildInputs = [

